### ISSUE TITLE
localStorage and three-cards

### ISSUE THREAD

**Issue by ilpiac** [Simbul/baker#719]
I think this is obviuos, but I've just realized it now :)
If I use three-cards rendering mode I have some problems with localStorage.
Obvious, as I've just said, because following and previous page are fully loaded while you're reading the actual page. So if you interact with actual page to set something with localStorage, this is not set when you swipe on the following or previous page.
In my case the reader can change the language on every page.
Really simple solution: screenshots rendering mode instead of three-cards.
But it's somewhat a pity because I really like three-cards, very smooth transitions and page already set :)
If - in future - is there a way to solve this (this is not a real problem/bug, I know) it would be really great.
Perhaps - but I'm not a programmer - swipe could trigger an event so that following/previous page could have a Javascript listener fot it (and so make checks/operations/calculations on localStorage)?

**Comment by folletto** [comment]
The events window `blur` and `focus` should be able to do what you are asking. Did you try? :)

**Comment by ilpiac** [comment]
It works like a charm, thanks Folletto! I didn't think about window.blur and window.focus, I'm definitively not a dev :(
Used window.focus - via JQuery - to check and set language via localStorage while using three-cards: really fine.
Just a little glitch: while swiping from the previous page to the actual one, the actual one shows "texts in old language" for half a second and then switches to the new language (if you've changed the language with localStorage in the previous page.)
Obvious: the new page needs some milliseconds to check localStorage and make some changes accordingly.

The best solution would be this one, because the listener makes changes every time localStorage changes, even if the window doesn't have focus:
<samp>
function onStorageEvent(storageEvent){
alert("storage event");
}
window.addEventListener('storage', onStorageEvent, false);
</samp>

It works inside Safari on OSX (listener works fine on pages inside tabs with no focus, different from the one that triggers the event and changes localStorage value).
But no way to have it working inside Baker. Perhaps the three pages - previous, actual and next one - are not like three tabs inside a browser session (I really don't know...)
I will go with the window.focus solution if I can't find if and how localStorage listeners work on Safari for iOS (I think that WebKit inside Baker and Safari for iOS are really the same thing, am I true?)

**Comment by ilpiac** [comment]
Made some tests with a server on a local network, using very same pages from Baker's book.
Safari running on OSX: listener for localStorage works fine with pages on different tabs.
Safari running on iOS: listener for localStorage works fine with pages on different tabs.
Baker app on iPad: listener for localStorage doesn't work between actual-previous-next pages.
Pages inside Baker work in a different way compared to tabs inside a Safari, perhaps every page inside Baker is like a separate browser instance and localStorage listener only works inside a single browser instance. 
I think I will go with <code>window.focus</code>, text flickering is not such a terrible thing :)

**Comment by philippgeuder** [comment]
you could use setInterval and check localStorage every 100ms or so ... see #198. ... proper events would be nice ;) those are tracked in #350

**Comment by ilpiac** [comment]
Thank you so much! Just read #198 and #350 and since every page in Baker is a different UIWebview is like using a listener for localStorage with different browsers: it simply doesn't work :)
The only way is yours, I have to check if it's better "text flickering" or checking something every second (even if there are only three pages - actual, previous and next - with this check running.)

**Comment by ilpiac** [comment]
I've used localStorage + setInterval to check it from side pages (previous and next).
Added window.focus and window.blur code to clear the interval (when page gains focus, in this case I don't need any more the check because language changes when the user click the language button) and to set the interval again (when page loses focus, so it can check again localStorage every second).
Not the very best way, but a way. I hope this is not too much resources-consuming :(
